api/src/methods/mod.rs

```diff
@@ -247,28 +248,28 @@ pub mod tests {
             }
         }
 
-        let block_hash_cache = SharedBlockHashCache::default();
+        let block_hash_cache = HybridBlockHashCache::new((), StubLatest(height));
 
         Box::new((
             ApplicationReader::<
                 TestDependencies<
+                    MockStateQueries,
                     _,
-                    InMemoryState,
                     _,
                     UmiBlockHash,
-                    _,
-                    SharedBlockHashCache,
-                    _,
+                    StubLatest,
+                    (),
+                    (),
+                    (),
+                    (),
+                    (),
+                    (),
+                    (),
+                    (),
+                    HybridBlockHashCache<(), StubLatest>,
+                    HybridBlockHashCache<(), StubLatest>,
```

Comment:
I wish this was less painful. Honestly I couldn't figure out why rustc failed to infer some type parameters and was perfectly fine with some others, so it was a guessing game when writing this.

Also as a side question, do you feel we would benefit from some more systematic mocking? We have this test dependencies instantiation in a few different places and it could be more nicely centralized.

Reply:
Yes, I completely agree. I had a hard time filling these in too. I think the core issue is that Rust's syntax isn't really meant to scale to this number of type generics. I'm not sure if there is a good way to make this more developer friendly.

Centralizing the testing implementations would be a welcome change! Probably lower priority than other things though.